proxy/src/config.rs

```diff
@@ -36,7 +36,24 @@ pub struct ProxyConfig {
     pub auth_link_uri: ApiUrl,
 }
 
-pub type TlsConfig = Arc<rustls::ServerConfig>;
+#[derive(Clone)]
+pub struct TlsConfig {
+    pub tls_config: Arc<rustls::ServerConfig>,
+    pub common_name: Option<&'static str>,
+}
+
+impl TlsConfig {
+    #[cfg(test)]
+    pub fn new(tls_config: Arc<rustls::ServerConfig>) -> TlsConfig {
+        TlsConfig {
+            tls_config,
+            common_name: None,
+        }
+    }
+    pub fn get_tls_config(self) -> Arc<rustls::ServerConfig> {
+        self.tls_config
+    }
```

Comment:
Sounds good, I'll make the change! Although I'm slightly confused because in the previous version of the code, we moved the tls_config, rather than cloning it, which is why this new access method now does the same. Also the compiler is not complaining about it. Let me know what you think about this reasoning! Maybe another solution is to simply rename the method move_tls_config, indicating that it's a move rather than a clone.
Regardless, your proposed changed sounds safer and doesnt make much of a performance difference.